### PR TITLE
Adds Vue component testing #10774

### PR DESCRIPTION
**Pull request by chrabyrd**
### Types of changes

-   [ ] Bugfix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Description of Change

Adds frontend testing to arches-core and projects.

MUST UPDATE `package.json` AND REBUILD `yarn.lock` ONCE https://github.com/archesproject/arches-dev-dependencies/pull/17 HAS BEEN MERGED

### Issues Solved

#10774

### CODE REVIEW

File: package.json
Comment (chrabyrd) [inline package.json:24]: This needs to be changed && yarn.lock rebuilt after merging https://github.com/archesproject/arches-dev-dependencies/pull/17

File: webpack/webpack.common.js
Comment (chrabyrd) [inline webpack/webpack.common.js:1]: This change needs to happen at the project level in tandem with moving the webpack directory to root

File: arches/management/commands/updateproject.py
Comment (jacobtylerwalls) [inline arches/management/commands/updateproject.py:109]: `updateproject` took a dive for me because the last line still expects the webpack template folder under install/, so if we're deleting that in this PR let's update the command, or maybe even better do it in #10758.

File: releases/7.6.0.md
Comment (jacobtylerwalls) [inline releases/7.6.0.md:68]: primevue seems major enough that if we're bumping to 3.51 let's update the list a few lines above here. (glad we're developing against latest, for sure!)

File: arches/install/arches-templates/package.json
Comment (jacobtylerwalls) [inline arches/install/arches-templates/package.json:10]: Do we need to rip out `--ext` here, too?

File: package.json
Comment (jacobtylerwalls) [inline package.json:11]: watch gives me warnings:

```node
(node:28231) ESLintIgnoreWarning: The ".eslintignore" file is no longer supported. Switch to using the "ignores" property in "eslint.config.js": https://eslint.org/docs/latest/use/configure/migration-guide#ignoring-files
```

File: releases/7.6.0.md
Comment (jacobtylerwalls) [inline releases/7.6.0.md:139]:
```suggestion
            "test": "vitest --run --coverage"
```

**Review by jacobtylerwalls — APPROVED** [review]
This rules. Thanks for digging in. Can't wait to start writing tests.

I assume the next chunk of work is a CI step, but I bet that's small enough to roll in now.

File: package.json
Comment (jacobtylerwalls) [reply package.json:11]: Also didn't see the instruction, sorry, resolving

File: arches/install/arches-templates/package.json
Comment (chrabyrd) [reply arches/install/arches-templates/package.json:10]: nope, this `--ext` is for nodemon

**Comment by chrabyrd** [comment]
@jacobtylerwalls this has been  updated to include frontend test coverage github actions. Please re-review it and its sister branch ( https://github.com/archesproject/arches-dev-dependencies/pull/17 ) at your convenience.

**Review by jacobtylerwalls — CHANGES_REQUESTED** [review]
With this change, I'm finding that a new 7.6 project can't build a frontend bundle because the webpack-stats.json appears at project root, i.e. next to node_modules, instead of at <project root>/webpack/webpack-stats.json.

LMK if you see something different.

**Comment by jacobtylerwalls** [comment]
> With this change, I'm finding that a new 7.6 project can't build a frontend bundle because the webpack-stats.json appears at project root, i.e. next to node_modules, instead of at /webpack/webpack-stats.json.

Never mind, cannot reproduce 👍

File: tsconfig.json
Comment (jacobtylerwalls) [inline tsconfig.json:17]: With this change, tsc no longer finds my type declarations like:

```ts
import type {
    ControlledList,
    ControlledListItem,
    NewItem,
} from "@/types/ControlledListManager";
```

giving instead:
```js
arches/app/src/components/ControlledListManager/utils.ts:7:8 - error TS2307: Cannot find module '@/types/ControlledListManager' or its corresponding type declarations.

7 } from "@/types/ControlledListManager";
```

I'm going to revert this change (and the related change in the templates and the updateproject.py) so that we can pull these changes in and unblock 8.0 work, but I would gladly re-review these changes in another a PR if you would like to discuss it futher.

File: package.json
Comment (jacobtylerwalls) [reply package.json]: done 👍

**Review by jacobtylerwalls — APPROVED** [review]
Glad to have this feature, and thanks for all the cleanups!

**Pull request merged**
